pkg/actions/cluster/unowned.go

```diff
@@ -21,3 +38,193 @@ func (c *UnownedCluster) Upgrade(dryRun bool) error {
 	_, err := upgrade(c.cfg, c.ctl, dryRun)
 	return err
 }
+
+func (c *UnownedCluster) Delete(waitTimeout time.Duration, wait bool) error {
+	clusterName := c.cfg.Metadata.Name
+
+	if err := c.checkClusterExists(clusterName); err != nil {
+		return err
+	}
+
+	clientSet, err := c.ctl.NewStdClientSet(c.cfg)
+	if err != nil {
+		return err
+	}
+
+	if err := deleteSharedResources(c.cfg, c.ctl, clientSet, waitTimeout); err != nil {
+		return err
+	}
+
+	// we have to wait for nodegroups to delete before deleting the cluster
+	// so the `wait` value is ignored here
+	if err := c.deleteAndWaitForNodegroupsDeletion(clusterName, waitTimeout); err != nil {
+		return err
+	}
+
+	if err := c.deleteIAMAndOIDC(wait, kubernetes.NewCachedClientSet(clientSet)); err != nil {
+		return err
+	}
+
+	return c.deleteCluster(clusterName, waitTimeout, wait)
+}
+
+func (c *UnownedCluster) checkClusterExists(clusterName string) error {
+	out, err := c.ctl.Provider.EKS().ListClusters(&awseks.ListClustersInput{})
+	if err != nil {
+		return err
+	}
+	for _, cluster := range out.Clusters {
+		if *cluster == clusterName {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("cluster %q not found", clusterName)
```

Comment:
`DescribeCluster` would be more appropriate:

```suggestion
	_, err := c.provider.EKS().DescribeCluster(&eks.DescribeClusterInput{
		Name: &c.spec.Metadata.Name,
	})
	if err != nil {
		if isNotFound(err) {
			return errors.Errorf("cluster %q not found", clusterName)
		}
		return errors.Wrapf(err, "error describing cluster %q", clusterName)
	}
	return nil
```

Reply:
good spot! I made the same mistake somewhere else and fixed it earlier, I forgot we could check if an err returned from `Describe` is a `notFound` error.  Thanks 😄